packages/contracts/contracts/Dependencies/TellorCaller.sol

```diff
@@ -43,11 +60,29 @@ contract TellorCaller is ITellorCaller {
             uint256 _timestampRetrieved
         )
     {
-        uint256 _count = tellor.getNewValueCountbyRequestId(_requestId);
-        uint256 _time =
-            tellor.getTimestampbyRequestIDandIndex(_requestId, _count.sub(1));
-        uint256 _value = tellor.retrieveData(_requestId, _time);
-        if (_value > 0) return (true, _value, _time);
-        return (false, 0, _time);
+        bytes memory _valueBytes;
+        (_valueBytes, _timestampRetrieved) = getDataBefore(_queryId, block.timestamp + 1);
```

Comment:
`getDataBefore` should be used with a time buffer to allow time for bad data to be disputed. Something like:

`getDataBefore(_queryId, block.timestamp - 15 minutes)`